force-app/main/default/classes/AffiliationMappingsService.cls

```diff
@@ -101,6 +101,29 @@ public virtual with sharing class AffiliationMappingsService {
         this.locateAffiliationMappingsMapper().updateAffiliationMappings(afflMappingModel);
     }
 
+    /*****************************************************************************
+     * @description Checks user permissions and update the auto enrollment mapppings.
+     *****************************************************************************/
+    public void updateAutoEnrollmentMappings(
```

Comment:
You're on the right path here. We do need to update both instances, but only need to perform one update if the new and old account record types match.

Reply:
As an additional note, if you have the name of the original mapping here, you can directly go to the mapper for that one since you know the relevant data points to update, name, status, and role.